snapcraft/plugins/catkin.py

```diff
@@ -127,77 +160,59 @@ def gcc_version(self):
     @property
     def rosdir(self):
         return os.path.join(self.installdir,
-                            'opt', 'ros', self.options.rosversion)
+                            'opt', 'ros', self.options.rosdistro)
 
-    def _deps_from_packagesxml(self, f, pkg):
-        tree = lxml.etree.parse(f)
+    def _find_dependencies(self):
+        if self.package_deps_found:
+            return
```

Comment:
Great question: I'm not sure (that was already in there). However, I *believe* it's because it's called from both `pull()` and `build()` (since both steps need to know it), but it doesn't want to find deps twice in case it's being run via `snapcraft` instead of `snapcraft <step>`.